src/servers/http_server.cc

```diff
@@ -269,8 +270,15 @@ ReadDataFromJsonHelper(
     TRITONSERVER_Error* assert_err =
         el.AssertType(triton::common::TritonJson::ValueType::ARRAY);
     if (assert_err == nullptr) {
-      RETURN_IF_ERR(ReadDataFromJsonHelper(base, dtype, el, counter));
+      RETURN_IF_ERR(
+          ReadDataFromJsonHelper(base, dtype, el, counter, expected_cnt));
     } else {
+      // Check if writing to 'serialized' is overrunning the expected byte_size
+      if (*counter == expected_cnt) {
```

Comment:
Use >=, counter for `TRITONSERVER_TYPE_BYTES` can increment more than 1 value. You probably need to add a similar check after line 380 after you know how much more bytes are needed to store the value (i.e. need `*counter + len + sizeof(uint32_t) <= expected_cnt`)

Reply:
Modified. Test cases for invalid request JSON are added. Please reveiw.